docs/ai_flags.md

```diff
@@ -0,0 +1,149 @@
+
+# How do you use them?
+Adding an AI flag to a trainer is straightforward, but the process is different depending on how trainers are being defined.
+
+## `COMPETITIVE_PARTY_SYNTAX == TRUE`
+If you are using competitive syntax parties, navigate to the trainer data in [`src/data/trainers.party`](https://github.com/rh-hideout/pokeemerald-expansion/blob/upcoming/src/data/trainers.party), find the trainer you’d like to change, and add flags like so:
+AI: Check Bad Move / Try to Faint / Check Viability. The name of each flag is just the constant, but without AI_FLAG at the beginning. For example, to add `AI_FLAG_SEQUENCE_SWITCHING`, any of the following will work:
+* AI_FLAG_SEQUENCE_SWITCHING
+* SEQUENCE_SWITCHING
+* SEQUENCE SWITCHING
+* Sequence_Switching
+* Sequence Switching
+
+## `COMPETITIVE_PARTY_SYNTAX != TRUE` / Not Found
+If you are not using competitive syntax parties, instead access the trainer data directly in [`src/data/trainers.h`](https://github.com/rh-hideout/pokeemerald-expansion/blob/master/src/data/trainers.h), and add flags like so, typed exactly the same as the flag names themselves: 
+`.aiFlags = AI_FLAG_CHECK_BAD_MOVE | AI_FLAG_TRY_TO_FAINT | AI_FLAG_CHECK_VIABILITY`
+
+# What AI Flags does pokeemerald-expansion have?
+This section lists all of expansion’s AI Flags and briefly describes the effect they have on the AI’s behaviour. In all cases, please check the corresponding function or surrounding code around their implementation for more details. Some of these functions are vanilla, some share a name with vanilla but have been modified to varying degrees, and some are completely new.
+
+## `AI_FLAG_CHECK_BAD_MOVE` 
+AI will avoid using moves that are considered poor in its current situation. The following list is not exhaustive. The AI will be disincentivized to:
+* Use powder moves into an immune target
+* Attack into invulnerability (Fly etc.) poorly
+* Use moves that are not very effective or worse
+* Use moves that are blocked or made worse by opposing abilities (don’t Poison Magic Guard, priority into Queenly Majesty, etc.) or partner abilities (Lightningrod redirect etc.)
+* Use moves that are negated by terrain
+* Use move effects that are poor (Attack buffs without a physical move, Charge if already charged, stat buffs if stats maxed, Roar when the opponent has one mon left, etc.)
+* Use moves that are not good in current weather, or set weather when partner is
+* Copy negative stat changes
+* Sucker Punch into predicted status move
+* Choice Lock into a poor move
```

Comment:
Instead of listing bullet points (which is just a subset of the functionality) I would summarize because the general gist of this flag is to prevent AI using moves that will fail, not nessasaty poor. How poor a move is is usually checked in `AI_FLAG_CHECK_VIABILITY`